src/atlas/grid/detail/grid/CubedSphere.h

```diff
@@ -241,9 +243,12 @@ class CubedSphere : public Grid {
     virtual std::string name() const override;
     virtual std::string type() const override;
 
-    // Return number of faces on cube
+    // Return N_, where (N_ * N_) is the number of cells on a tile
     inline idx_t N() const { return N_; }
 
+    // Return the areas of the faces/cells
+    Field gridCellArea(const FunctionSpace&) const;
```

Comment:
In order to encapsulate as much as possible the grid from objects like Field, FunctionSpace, Mesh,
I would avoid this function.

Possibly you could create a different function
```
double gridCellArea(idx_t index);
```
The grid itself should have all possible information needed to compute just a single cell value.

Then where needed fill a container such as `std::vector` or `atlas::Field` or ...
But in this case here, I think you could just compute the cellArea within the loop where you compute the weights, avoiding the entire allocation of a container.

Reply:
@wdeconinck  ... good point. However, there is reason why the function `CubedSphere::gridCellArea()` has been designed and implemented in this way.

So far, at the MO, we have identified at least two workflow steps where this function is needed: one is the evaluation of the _area weights matrix_ within the interpolation/binning procedure (this PR); the other one is the evaluation of the _Jc term_, a penalty term associated with the gravity wave constraints for the cost function.
Thus, we think that having a function signature for `CubedSphere::gridCellArea()` based on the `FunctionSpace` data type make it general enough to be used in different contexts.
<br>

Reply:
I managed to boil this down to this standalone function, independent of functionspace
```c++
// Area around nodes of cubed sphere mesh
Field CubedSphereNodalArea(const Mesh& mesh) {

  constexpr auto deg2rad = M_PI / 180.;
  const auto& proj = mesh.projection();
  auto lonlat = array::make_view<double, 2>(mesh.nodes().lonlat());
  auto gcell_area_field = Field("grid_cell_areas", make_datatype<double>(), array::make_shape(mesh.nodes().size()));  
  auto gcell_area_fview = array::make_view<double, 1>(gcell_area_field);

  double gcell_area_cs = [&] {
      /******* CUBED_SPHERE SPECIFIC *********/
      // (grid_res * grid_res) = no. of cells on a tile
      ATLAS_ASSERT(CubedSphereGrid(mesh.grid()));
      auto grid_res = CubedSphereGrid(mesh.grid()).N();
      // area of a grid cell (cubed-sphere coord. system)
      return M_PI/(2*grid_res) * M_PI/(2*grid_res);
   }();

  for (size_t i = 0; i < gcell_area_fview.size(); i++) {
    PointLonLat loc = PointLonLat(lonlat(i, atlas::LON), lonlat(i, atlas::LAT));
    double cos_lat = std::cos(deg2rad * loc.lat());
    double grid_jac_det = 1/proj.jacobian(loc).determinant();
    // area of a grid cell (geographic coord. system)
    gcell_area_fview(i) = grid_jac_det * gcell_area_cs * cos_lat ;   /****** USE gcell_area_cs here ******/
  }

  return gcell_area_field;
}
```
Possibly with more thought this could be made even more independent of CubedSphere.


What happens to the Binning procedure if the combination does not happen to be "grid == CubedSphere && functionspace == NodeColumns" ?

Reply:
@wdeconinck ... thanks for this.
Where do you think  this function should be stored? In which file, class,...?

Reply:
It could be a mesh-action, similar to https://github.com/ecmwf/atlas/blob/develop/src/atlas/mesh/actions/BuildCellCentres.h 
So putting it parallel to this file, and following its design is a good idea.
You can then "register" this field within the mesh.nodes() and could be reused later.

What happens to the Binning procedure if the combination does not happen to be "grid == CubedSphere && functionspace == NodeColumns" ?

Reply:
> functionspace == NodeColumns

The _binning_ procedure works also for "functionspace == StructuredColumns" (grid type: "O").
I implemented a test to check this scenario, see [here](https://github.com/JCSDA-internal/atlas/blob/feature/interpolation_binning_01/src/tests/interpolation/test_interpolation_binning.cc#L155).